### PR TITLE
Final vals

### PR DESCRIPTION
**Pull request by jporemba**

### CODE REVIEW

File: src/main/scala/com/cluster/graph/ClusterShardingApp.scala [hunk 1]
```diff
@@ -168,6 +175,33 @@ object ClusterShardingApp {
...
+    // Wait until finished
+    
+
+    var finalVals: Map[Int, Option[Colour]] = null
+    while(null == finalVals){
+      Thread.sleep(1000)
+
+      val timeout: Timeout = 5.seconds
+      val sched = entityManager.scheduler
+      val future: Future[GlobalCoordinator.FinalValuesResponse] = gcRef.ask(ref => GlobalCoordinator.GetFinalValues(ref))(timeout,sched)
```
Comment (atrostan) [inline src/main/scala/com/cluster/graph/ClusterShardingApp.scala:188]: Ahh. I assume this was the source of the bug? We were trying to access the final value map before all the data was actually received?
Good fix -- it should be noted that a valueMap may be quite large: M/Billions of vertices? -- another solution would be that each PartitionCoordinator persists their (final) value maps to their local machine. But this is good for now!

File: src/main/scala/com/cluster/graph/GlobalCoordinator.scala [hunk 2]
```diff
@@ -135,6 +172,11 @@ object GlobalCoordinator {
   final case class DONE(stepNum: Int) extends Command
   final case class TerminationVote(stepNum: Int) extends Command
   final case class BEGIN() extends Command
+  /**
+   * JsonDeserialize tag needed. See open issue: https://github.com/akka/akka/issues/28566
```
Comment (atrostan) [inline src/main/scala/com/cluster/graph/GlobalCoordinator.scala:176]: Fun.

**Review by atrostan — COMMENTED** [review]
LGTM!

File: src/main/scala/com/cluster/graph/ClusterShardingApp.scala (hunk 1 above)
Comment (jporemba) [reply src/main/scala/com/cluster/graph/ClusterShardingApp.scala:188]: Actually the bug is the JsonDeserialize issue mentioned below. This is just to get to the map from the GC to the main app.

And yeah, I agree this is not necessarily a good long term solution - just to get something working!

**Pull request merged**